concordium-std/src/impls.rs

```diff
@@ -697,21 +716,18 @@ where
     }
 
     pub fn is_empty(&self) -> bool {
-        if let Some(_) = self.state_ll.borrow().iterator(&self.prefix).next() {
-            false
-        } else {
-            true
-        }
+        let iterator = self.state_ll.borrow().iterator(&self.prefix);
+        iterator.is_err() || iterator.unwrap().next().is_none()
```

Comment:
`iterator(..)` will return an error if you have more than `N` iterators for the same key. 
So if you get `N` iterators for `self.prefix` and then call `is_empty()`, it will return `true` because you will get an error (despite the trie not being empty).

Reply:
Since `N=2^16` perhaps we should just abort in that case?

Reply:
Looking at the details of the implementation in the host now, there seems to be no disadvantage in using a u32 for this limit, so I think we can raise it to 2^32, which makes using abort here appropriate.